packages/gatsby-theme-docs/src/plugins/rehype-id-slug.js

```diff
@@ -0,0 +1,26 @@
+const visit = require('unist-util-visit');
+const hasProperty = require('hast-util-has-property');
+const headingRank = require('hast-util-heading-rank');
+const toString = require('hast-util-to-string');
+const Slugger = require('github-slugger');
+
+const slugs = new Slugger();
+
+/**
+ * This plugins is a re-implementation of rehype-slug https://github.com/rehypejs/rehype-slug
+ * The purpose of the plugin is to add id to headings based on the text content.
+ * This re-implementation simply ensure that the ids get postProcessed by a custom `postProcess` function, the
+ * rest of the implementation works exactly like the original.
+ */
+module.exports =
+  (options = {}) =>
+  (tree) => {
+    slugs.reset();
+    const postProcess = options.postProcess || ((input) => input);
+    visit(tree, 'element', (node) => {
+      if (headingRank(node) && node.properties && !hasProperty(node, 'id')) {
+        const nodeSlug = slugs.slug(toString(node));
```

Comment:
```suggestion
        const nodeSlug = slugs.slug(toString(node).trim());
```
trim could actually be enoug. 

I suggest to move the slugs.slug()  call into the postprocessing, too to really make sure the slugging is the same.  more like our own slug function